terraso_backend/apps/core/models/groups.py

```diff
@@ -40,11 +41,13 @@ class Group(SlugModel):
 
     MEMBERSHIP_TYPE_OPEN = "open"
     MEMBERSHIP_TYPE_CLOSED = "closed"
+    MEMBERSHIP_TYPE_RESTRICTED = "restricted"
```

Comment:
Suggestion: From what I understand it seems that `restricted` is not the type of the group, it is the mechanism available to join the group, and that can have different variations:
1. User can join a group
2. User can be added to a group
3. User can join or be added to a group

From the conversation we had the other day, what we want to have in the long term is a data entity that is separated from group, landscape and project that handles the logic related to memberships. So, from that I understand that maybe having all of that now is a big change, but I think we should move a step closer to that long term goal, and from that point of view, we need a way to define how the user can join the group, maybe we can have something called `join_method`?

Reply:
> my understanding of the code is that as it is, if we make project groups "open" the previous API surface would allow manager or user to add a membership.

Yep, correct. I think this would almost work if we set groups closed, but the way it works right now is that the managers of the group are sent an email asking to confirm membership after the user is added (you can check out the `EditMembershipMutation` to see the logic). Unfortunately I think that would be really annoying for the manager to have to go to their email to confirm users they are adding.

Reply:
hmm OK, what if we just make them "open" for now and then we will close that hole up with the refactor long before we actually release anything?

Reply:
@david-code I think that ideally the type of the group should not have anything to do with how the members join the group, type is more related to the data related to a group, and specific requirements linked in our case to 1000L group, landscape and project, right now it is working with closed because there were no other use cases, but now we have enough requirements to make this distiction, so I think we should remove the `restricted` type and just add a new field that will define the join behavior `join_method` that way we dont have to define a new name for a type everytime we need a new combination between (access_data(type?), user_can_join, can_invite_user)